packages/@webex/plugin-meetings/src/multistream/mediaRequestManager.ts

```diff
@@ -90,14 +100,48 @@ export class MediaRequestManager {
     this.slotsActiveInLastMediaRequest = activeSlots;
   }
 
+  public setDegradationPreferences(degradationPreferences: DegradationPreferences) {
+    this.degradationPreferences = degradationPreferences;
+    this.sendRequests(); // re-send requests after preferences are set
+  }
+
+  private getDegradedClientRequests() {
+    const clientRequests = {...this.clientRequests};
+    const maxFsLimits = [8192, 3600, 920, 240, 60]; // 1080p, 720p, 360p, 180p, 90p
+
+    // reduce max-fs until total macroblocks is below limit
+    for (let i = 0; i < maxFsLimits.length; i += 1) {
+      let totalMacroblocksRequested = 0;
+      Object.values(clientRequests).forEach((mr) => {
+        if (mr.codecInfo) {
+          mr.codecInfo.maxFs = Math.min(
+            mr.codecInfo.maxFs || CODEC_DEFAULTS.h264.maxFs,
+            maxFsLimits[i]
+          );
+          totalMacroblocksRequested += mr.codecInfo.maxFs * mr.receiveSlots.length;
+        }
+      });
+      if (totalMacroblocksRequested <= this.degradationPreferences.maxMacroblocksLimit) {
+        if (i !== 0) {
+          LoggerProxy.logger.warn(
+            `multistream:mediaRequestManager --> too many requests with high max-fs, frame size may be limited`
+          );
+        }
+        break;
+      }
+    }
```

Comment:
in the case when applying even the last value from `maxFsLimits` (60) doesn't put the total under the `maxMacroblocksLimit`, we just end up sending the requests anyway with that value. That's not entirely correct, although probably OK in reality as we will usually have a very high `maxMacroblocksLimit`, but maybe worth printing a console warning? or you could add 0 as the last value in `maxFsLimits`

Reply:
Is 0 even a valid maxFs value?

I kind of just assumed that it's ok to have as many 90p streams as the user wants, but I guess that's not a good assumption. Will add an additional warning message, maybe something like "you are requesting too many streams, consider reducing the number of requests".

Reply:
Is 0 even a valid maxFs value?

I kind of just assumed that it's ok to have as many 90p streams as the user wants, but I guess that's not a good assumption. Will add an additional warning message, maybe something like "you are requesting too many streams, consider reducing the number of requests".

Reply:
yeah good point, not sure if 0 would be accepted by WCME or Homer. I think it's worth covering the edge case for even 90p not being small enough just in case someone is testing this stuff with very low values of `maxMacroblocksLimit`.